pythia/seo/tablematcher.py

```diff
@@ -0,0 +1,201 @@
+            feature_1 = df_1.columns.values
+
+        if feature_2 is None:
+            feature_2 = df_2.columns.values
+
+        if len(feature_1) != len(feature_2):
+            raise SunpyUserWarning(
+                "The number of columns to match the rows on must be the same.")
+
+        try:
+            df_1 = df_1[feature_1]
+        except KeyError:
+            raise SunpyUserWarning("The features specified for table 1 do not "
+                                   "correspond to any columns in table 1.")
+        try:
+            df_2 = df_2[feature_2]
+        except KeyError:
+            raise SunpyUserWarning("The features specified for table 2 do not "
+                                   "correspond to any columns in table 2.")
+
+        return df_1, df_2
+
+    def match_cosine(self, df_1, df_2):
+        """
+        Finds Cosine similarity between the rows of the two dataframes.
+        Parameters
+        ----------
+        df_1: `pd.DataFrame`
+            First DataFrame to match the rows from.
+        df_2: `pd.DataFrame`
+            Second DataFrame to match the rows from.
+
+        Returns
+        -------
+        result: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains indices of rows from df_2 that best correspond to rows from df_1.
+        match_score: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains match score for  corresponding best matches.
+        """
+
+        try:
+            from sklearn.metrics.pairwise import cosine_similarity
+        except ImportError:
+            raise SunpyUserWarning(
+                "Table Matcher requires Scikit Learn to be installed")
+
+        cosine = cosine_similarity(X=df_1, Y=df_2)
+        result = np.argmax(cosine, axis=1)
+        match_score = np.max(cosine, axis=1)
+
+        return result, match_score
+
+    def match_euclidean(self, df_1, df_2):
+        """
+        Finds euclidean distance between the rows of the two dataframes.
+        Parameters
+        ----------
+        df_1: `pd.DataFrame`
+            First DataFrame to match the rows from.
+        df_2: `pd.DataFrame`
+            Second DataFrame to match the rows from.
+        Returns
+        -------
+        result: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains indices of rows from df_2 that best correspond to rows from df_1.
+        match_score: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains match score for  corresponding best matches.
+        """
+        try:
+            from sklearn.metrics.pairwise import euclidean_distances
+        except ImportError:
+            raise SunpyUserWarning(
+                "Table Matcher requires Scikit Learn to be installed")
+
+        euclidean = euclidean_distances(X=df_1, Y=df_2)
```

Comment:
from the tests, `df_1` and `df_2` can't be compared, one has units and the other hasn't. So, somewhere we need to make sure both share the same units.